spec/controllers/orgs/group_assignment_repos_controller_spec.rb

```diff
@@ -6,26 +6,19 @@
   let(:user)         { classroom_teacher }
   let(:organization) { classroom_org     }
   let(:student)      { classroom_student }
-
   let(:repo_access)  { RepoAccess.create(user: student, organization: organization) }
-
   let(:group_assignment) do
     create(:group_assignment, title: "Learn Ruby", organization: organization, public_repo: false)
   end
-
-  let(:group) { Group.create(title: Time.zone.now, grouping: group_assignment.grouping) }
-
-  let(:group_assignment_repo) do
-    GroupAssignmentRepo.create(group_assignment: group_assignment, group: group)
-  end
+  let(:group) { create(:group, grouping: group_assignment.grouping, github_team_id: 2_976_561) }
+  let(:group_assignment_repo) { GroupAssignmentRepo.create(group_assignment: group_assignment, group: group) }
```

Comment:
Will using an actual `GroupAssignmentRepo` instead of a factory cause any flaky-ness? That might be the next thing to tackle if it also causes problems

Reply:
A factory would be better. Unfortunately the `GroupAssignmentRepo` model suffers from the same problem that the `Group` model suffers from: networks requests in the validations.

I would love to clean that up and move `GroupAssignmentRepo` to a `Creator` pattern and clean up its job!